homeworks/Maria.Priamuhina_Helen-Mary/Homework3/htask1.js

```diff
@@ -0,0 +1,60 @@
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+});
+
+/* task1.1 */
+function Square(num) {
+    return num * num;
+}
+
+rl.on('line', (input) => {
+    const num = input;
+    rl.close();
+    console.log(Square(num));
+});
+
+/* task 1.2 */
+function CreateArray(end, step) {
+    const array = [0];
+    const newEnd = Number(end);
+    const newStep = Number(step);
+    let temp = 0;
+    for (let i = 1; (temp + newStep) <= newEnd; i += 1) {
+        array[i] = temp + newStep;
+        temp += newStep;
+    }
+    for (let i = 0; i < array.length; i += 1) { console.log(array[i]); }
+    return array;
+}
+
+/* task 1.4 */
+function SumOfElems(array) {
+    let sum = 0;
+    for (let i = 0; i < array.length; i += 1) {
+        sum += array[i];
+    }
+    console.log(sum);
+}
+/* task 1.3 */
+function SquareArray(array) {
+    for (let i = 0; i < array.length; i += 1) {
+        const temp = Square(array[i]);
+        // eslint-disable-next-line no-param-reassign
+        array[i] = temp;
```

Comment:
code for tasks 3 and 4 is working, but may be significantly optimized with `map()`, `reduce()`, `push()` methods